File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/diagnosiskeydist/Submission.java
package uk.nhs.nhsx.diagnosiskeydist;

import uk.nhs.nhsx.diagnosiskeyssubmission.model.StoredTemporaryExposureKeyPayload;

import java.time.Instant;

public class Submission {

    public final Instant submissionDate;
    public final StoredTemporaryExposureKeyPayload payload;

    public Submission(Instant submissionDate, StoredTemporaryExposureKeyPayload payload) {
        this.submissionDate = submissionDate;
        this.payload = payload;
    }

}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/diagnosiskeydist/Submission.kt
package uk.nhs.nhsx.diagnosiskeydist

import uk.nhs.nhsx.diagnosiskeyssubmission.model.StoredTemporaryExposureKeyPayload
import java.time.Instant

data class Submission(
    val submissionDate: Instant,
    val payload: StoredTemporaryExposureKeyPayload
)
